### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by leshihak**

### CODE REVIEW

**Comment by tronya** [comment]
Screenshot 2022-02-22 at 09 12 23
For some reason user can not be defined in home page

**Comment by tronya** [comment]
> Screenshot 2022-02-22 at 09 12 23 For some reason user can not be defined in home page

Sorry not on the home page, its on charts page, when users stays on the page some time.

**Comment by tronya** [comment]
Screenshot 2022-02-22 at 09 15 18
And some another bug here!

File: pages/chats/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    .filter((user) => authUser.uid !== user.uid)
+    .sort((a, b) =>
+      a.displayName!! < b.displayName!!
+        ? -1
+        : b.displayName!! > a.displayName!!
+        ? 1
+        : 0
+    )
+    .filter((person) =>
+      person.displayName!!.toLowerCase().includes(searchTerm.toLowerCase())
```
Comment (tronya) [inline pages/chats/index.tsx]: Here should be a check for display name exists, like 
person.displayName ? person.displayName.toLowerCase().includes(searchTerm.toLowerCase()) : false

File: pages/chats/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  const { user: authUser } = useAuth();
+  const usersGeolocation = useUsersGeolocation();
+  const router = useRouter();
+  const { screenType } = useWindowDimensions();
+
+  if (!usersGeolocation || !authUser) {
+    return <Loader />;
+  }
+
+  const users = usersGeolocation
```
Comment (tronya) [inline pages/chats/index.tsx:34]: Maybe its better to use
[https://firebase.google.com/docs/auth/admin/manage-users#list_all_users](url) to fetch all existed users, not only thosee that we store. but (could be disscused)

**Comment by tronya** [comment]
Screenshot 2022-02-22 at 10 31 56
Icantclose modal of emoji , only submiting form

**Comment by leshihak** [comment]
> Screenshot 2022-02-22 at 09 15 18 And some another bug here!

fixed it in another PR https://github.com/tronya/tronya.github.io/pull/45

**Comment by leshihak** [comment]
> Screenshot 2022-02-22 at 10 31 56 Icantclose modal of emoji , only submiting form

Modal closes\opens on Emoji icon

**Review by tronya — APPROVED** [review]

**Pull request merged**